x-pack/plugins/viz_editor/public/common/components/draggable/draggable.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/**
+ * A basic drag / drop helper. Modifies a global "value" to track what value is being dragged
+ * around.
+ *
+ * TODO: Replace this with something more robust...
+ */
+
+import React, { useState } from 'react';
+
+interface Props {
+  value?: any;
+  className?: string;
+  activeTargetClassName?: string;
+  draggable?: boolean;
+  canHandleDrop?: (value: any) => boolean;
+  onDrop?: (value: any) => void;
+  children: any;
+}
+
+// HACK: Track what value is being dragged... ultimately, this'll be
+// managed by EUI, so maybe a mutable global is OK for now? :/
+let draggingValue: any;
```

Comment:
Do we have to think about how this will work for multiple plugins which can't share a mutable global (do we have to put it on the window object)? Maybe we should talk with the EUI team on whether and how their solution will support such scenarios

Reply:
This is not a permanent file. Just a hack. It sounds like EUI's solution will actually be available sooner than I expected (a few days). But yeah. I think we'd have to put it on the window, probably if plugins can't share the module.